ir/x86_intrinsics.h

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include "ir/instr.h"
+
+namespace IR {
+class FakeShuffle final : public Instr {
+  Value *v1, *v2, *mask;
+
+public:
+  FakeShuffle(Type &type, std::string &&name, Value &v1, Value &v2, Value &mask)
+      : Instr(type, std::move(name)), v1(&v1), v2(&v2), mask(&mask) {}
+  std::vector<Value *> operands() const override;
+  bool propagatesPoison() const override;
+  bool hasSideEffects() const override;
+  void rauw(const Value &what, Value &with) override;
+  void print(std::ostream &os) const override;
+  StateValue toSMT(State &s) const override;
+  smt::expr getTypeConstraints(const Function &f) const override;
+  std::unique_ptr<Instr> dup(Function &f,
+                             const std::string &suffix) const override;
+};
+
+class X86IntrinBinOp final : public Instr {
+public:
+  static constexpr unsigned numOfX86Intrinsics = 135;
+  enum Op {
+#define PROCESS(NAME, A, B, C, D, E, F) NAME,
+#include "x86_intrinsics_binop.inc"
+#undef PROCESS
+  };
+
+  // the shape of a vector is stored as <# of lanes, element bits>
+  static constexpr std::array<std::pair<unsigned, unsigned>, numOfX86Intrinsics>
```

Comment:
these should be moved to the C++ file (static globals) since they don't have to be publicly exposed.